From f81cdbbbdb85e3032f4f18ca43f491e6888c1e0a Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Fri, 3 Jun 2016 00:31:20 +0200
Subject: [PATCH 06/20] Declare all phony targets for efficiency. Remove call
 to dh_clean (done by dh even if dh_auto_clean is overriden). Only mkdir or CC
 when necessary. Warn for unexpected targets instead of running dh.

---
 debian/rules | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/debian/rules b/debian/rules
index 7e3e584..020cca6 100755
--- a/debian/rules
+++ b/debian/rules
@@ -4,20 +4,25 @@ DPKG_EXPORT_BUILDFLAGS = 1
 include /usr/share/dpkg/buildflags.mk
 include /usr/share/ada/debian_packaging*.mk
 
-%:
-	dh ${@}
+POLICY_TARGETS := binary binary-arch binary-indep build build-arch \
+  build-indep clean
+.PHONY: $(POLICY_TARGETS)
+$(POLICY_TARGETS):
+	dh $@
+
+# Prevent debhelper executing any upstream Makefile target.
+.PHONY: $(addprefix override_dh_auto_, configure build test install clean)
 
 override_dh_auto_clean:
 	rm -rf obj adabrowse sdefault.adb ad-projects-impl_yes-get_parent.adb
-	dh_clean
 
-override_dh_auto_build:
-	mkdir -p obj
-	$(CC) -c util-nl.c -o obj/util-nl.o $(CFLAGS) $(CPPFLAGS)
+override_dh_auto_build: obj/util-nl.o
 	gnatmake $(BUILDER_OPTIONS) -Pdebian/misc/adabrowse.gpr
+obj/util-nl.o: util-nl.c | obj
+	$(CC) -c $< -o $@ $(CFLAGS) $(CPPFLAGS)
+obj:
+	mkdir $@
 
-override_dh_auto_test:
-override_dh_auto_install:
-
+.PHONY: override_dh_compress
 override_dh_compress:
 	dh_compress -X.ads -X.adb
-- 
2.8.1

